state/aws.go

```diff
@@ -35,7 +35,9 @@ func NewAWS(c *config.Config) AWS {
 
 	if len(c.AWS.APPRoleArn) > 0 {
 		log.Debugf("Using %s role", c.AWS.APPRoleArn)
-		creds := stscreds.NewCredentials(sess, c.AWS.APPRoleArn)
+		creds := stscreds.NewCredentials(sess, c.AWS.APPRoleArn, func(p *stscreds.AssumeRoleProvider) {
+			p.ExternalID = aws_sdk.String(c.AWS.ExternalID)
```

Comment:
What happens when the `ExternalID` is not specified? Does it behave the same as today?

Reply:
It should be the same, as the default value is always an empty string. Moreover, let me do some tests to verify it, and I will come back to you! 😄 

thanks